geode-core/src/main/java/org/apache/geode/management/internal/cli/functions/RegionFunctionArgs.java

```diff
@@ -459,81 +447,72 @@ public EvictionAttributes getEvictionAttributes() {
-    private static final long serialVersionUID = 1474255033398008062L;
+    private static final long serialVersionUID = 1474255033398008063L;
 
-    private final ExpirationFor type;
-    private final ExpirationAttributes timeAndAction;
+    private final Integer time;
+    private final ExpirationAction action;
 
-    public ExpirationAttrs(ExpirationFor type, Integer time, String action) {
-      this.type = type;
-      if (time != null) {
-        this.timeAndAction = new ExpirationAttributes(time, getExpirationAction(action));
-      } else {
-        this.timeAndAction = new ExpirationAttributes(0, getExpirationAction(action));
-      }
+    public ExpirationAttrs(Integer time, ExpirationAction action) {
+      this.time = time;
+      this.action = action;
     }
 
-    public ExpirationAttributes convertToExpirationAttributes() {
-      return timeAndAction;
-    }
-
-    /**
-     * @return the type
-     */
-    public ExpirationFor getType() {
-      return type;
-    }
-
-    /**
-     * @return the time
-     */
     public Integer getTime() {
-      return timeAndAction.getTimeout();
+      return time;
     }
 
-    /**
-     * @return the action
-     */
     public ExpirationAction getAction() {
-      return timeAndAction.getAction();
+      return action;
     }
 
     @Override
-    public String toString() {
-      StringBuilder builder = new StringBuilder();
-      builder.append(ExpirationAttrs.class.getSimpleName() + " [type=");
-      builder.append(type);
-      builder.append(", time=");
-      builder.append(timeAndAction.getTimeout());
-      builder.append(", action=");
-      builder.append(timeAndAction.getAction());
-      builder.append("]");
-      return builder.toString();
-    }
-
-    private static ExpirationAction getExpirationAction(String action) {
-      if (action == null) {
-        return ExpirationAttributes.DEFAULT.getAction();
+    public boolean equals(Object object) {
+      if (object == null) {
+        return false;
       }
-      action = action.replace('-', '_');
-      if (action.equalsIgnoreCase(ExpirationAction.DESTROY.toString())) {
-        return ExpirationAction.DESTROY;
-      } else if (action.equalsIgnoreCase(ExpirationAction.INVALIDATE.toString())) {
-        return ExpirationAction.INVALIDATE;
-      } else if (action.equalsIgnoreCase(ExpirationAction.LOCAL_DESTROY.toString())) {
-        return ExpirationAction.LOCAL_DESTROY;
-      } else if (action.equalsIgnoreCase(ExpirationAction.LOCAL_INVALIDATE.toString())) {
-        return ExpirationAction.LOCAL_INVALIDATE;
-      } else {
-        throw new IllegalArgumentException(
-            CliStrings.format(CliStrings.CREATE_REGION__MSG__EXPIRATION_ACTION_0_IS_NOT_VALID,
-                new Object[] {action}));
+
+      if (!(object instanceof ExpirationAttrs)) {
+        return false;
       }
+
+      ExpirationAttrs that = (ExpirationAttrs) object;
+      return time == that.time && action == that.action;
```

Comment:
Do you mean to use `.equals` and not `==` here?  Otherwise, this just boil down to the `Object.equals` and can inherited.

Reply:
ExpirationAction is an object behaving like an Enum. The constructor is private.  All instances are predefined. See the isXXX method in this object

  public boolean isLocalDestroy() {
    return this == LOCAL_DESTROY;
  }